### DISCUSSION TITLE
Come testare webhooks in fase di sviluppo

### DISCUSSION THREAD

**Discussion by FrancescoAntonelli** [fattureincloud/api#381] [Q&A] [answered]
Salve,
  sto cercando di implementare un'integrazione tramite webhook ma ho il problema di poter fare dei test sulla mia macchina in fase di sviluppo. Sono per forza obbligato a pubblicare la mia app su un sito https per vedere l'arrivo degli eventi? Non c'è una maniera tale per cui possa utilizzare la mia macchina come server?

Grazie mille
  Francesco

**Comment by emajo** [comment, marked as answer]
Ciao @FrancescoAntonelli, in fase di test per vedere gli esempi il mio consiglio è quello di utilizzare servizi come https://webhook.site, in quanto per utilizzare la sua macchina il procedimento è abbastanza complicato in quanto implica del port forwarding e l'aggiunta di un certificato tls.

Grazie,
Emanuele.

**Reply by FrancescoAntonelli** [reply]
Grazie @emajo sto cercando di usare il sito suggerito solo che riesco a sottoscrivermi agli eventi e ricevere la chiamata di verifica della sottoscrizione, solo che poi dalla GET il return della mia webApi non arriva più verso di voi e "muore" su webhook.site
e quindi non riesco a trovare la sottoscrizione "verificata". Qualche suggerimento?
Grazie
  Francesco

**Reply by valmoz** [reply]
Buongiorno @FrancescoAntonelli,

le suggerisco due metodi per verificare la sottoscrizione:
1. Tramite webhook.site: In alto a destra è presente la sezione "edit" che permette di impostare la risposta automatica. Nell'immagine seguente le mostro come usare la variabile per effettuare la validazione

![image](https://github.com/fattureincloud/api/assets/7254293/ffac8e36-f118-4335-9f5a-7b7ab66ab720)

2. Rilasciare temporaneamente la sua applicazione (o una piccola applicazione con il solo codice necessario) su qualche servizio come [Heroku](https://www.heroku.com/home), [Google AppEngine](https://cloud.google.com/appengine?hl=it) o simili

Se invece volesse testare il codice della validazione della sottoscrizione in locale, le suggerirei di simulare la chiamata in ingresso tramite [Postman](https://www.postman.com/) o app simili, e di effettuare la chiamata verso localhost.

<img width="913" alt="image" src="https://github.com/fattureincloud/api/assets/7254293/2729b510-db2e-44fa-83f7-55c1016fe392">

Ovviamente, se vuole, può usare Postman anche per verificare che webhook.site risponda come atteso alla richiesta di verifica, impostando l'URL proposto dall'applicazione.

A disposizione
Mauro

**Reply by FrancescoAntonelli** [reply]
Grazie mille, la prima soluzione ha funzionato alla perfezione :)
